app/server/auth/authenticate.py

```diff
@@ -12,6 +12,16 @@
 from app.config.config import settings
 
 
+@router.get("/gitlab/login")
+async def login_gitlab():
+    APP_ID, REDIRECT_URI = (
+        settings.GITLAB_CLIENT_ID,
+        "http://localhost:8080/api/auth/callback/gitlab",
+    )
+    url = f"https://gitlab.com/oauth/authorize?client_id={APP_ID}&redirect_uri={REDIRECT_URI}&response_type=code"
+    return RedirectResponse(url=url)
+
+
 @router.get("/api/auth/callback/gitlab")
```

Comment:
Can we rename this to `/gitlab/token` instead? There's no uniformity in the API routes.

So we can have `/github/token` and `/gitlab/token`? Then split both routes so on the Swagger page they can be differentiated. Example:

# Token

/gitlab/token
/github/token

# Policy Operation

/policies*

# Data operation

/ data

# Repo management


/gitlab/repo
/github/repo

---

As we progress later on, we can update the functions so we just have a single route with an extra parameter `git_provider` that enables the function/method return the right output.